test/plugins/validation/2and3/schema-ibm.js

```diff
@@ -1005,8 +882,152 @@ describe('validation plugin - semantic - schema-ibm - OpenAPI 3', () => {
+    );
+
+    const res = validate({ jsSpec: spec, isOAS3: true }, config);
+    expect(res.errors.length).toEqual(3);
+    expect(res.errors[0].message).toEqual('oneOf should be an array');
+    expect(res.errors[0].path).toEqual([
+      'components',
+      'schemas',
+      'complexOneOfError',
+      'oneOf',
+      0,
+      'oneOf'
+    ]);
+    expect(res.errors[1].message).toEqual('allOf should be an array');
+    expect(res.errors[1].path).toEqual([
+      'components',
+      'schemas',
+      'complexAllOfError',
+      'oneOf',
+      0,
+      'allOf'
+    ]);
+    expect(res.errors[2].message).toEqual('anyOf should be an array');
+    expect(res.errors[2].path).toEqual([
+      'components',
+      'schemas',
+      'complexAnyOfError',
+      'oneOf',
+      0,
+      'anyOf'
+    ]);
+  });
+
+  it('should report an error when non-array allOf, anyOf, or oneOf schema used in object prop', () => {
+    const spec = yaml.safeLoad(
+      fs.readFileSync(
+        'test/cli-validator/mockFiles/oas3/composeModelProps.yaml'
+      )
+    );
+
+    const res = validate({ jsSpec: spec, isOAS3: true }, config);
+    expect(res.errors.length).toEqual(3);
+    expect(res.errors[0].message).toEqual('oneOf should be an array');
+    expect(res.errors[0].path).toEqual([
+      'components',
+      'schemas',
+      'object1',
+      'properties',
+      'one_of_error_prop',
+      'schema',
+      'oneOf'
+    ]);
+    expect(res.errors[1].message).toEqual('allOf should be an array');
+    expect(res.errors[1].path).toEqual([
+      'components',
+      'schemas',
+      'object1',
+      'properties',
+      'all_of_error_prop',
+      'schema',
+      'allOf'
+    ]);
+    expect(res.errors[2].message).toEqual('anyOf should be an array');
+    expect(res.errors[2].path).toEqual([
+      'components',
+      'schemas',
+      'object1',
+      'properties',
+      'any_of_error_prop',
+      'schema',
+      'anyOf'
+    ]);
+  });
+
+  it('should report an error when non-array allOf, anyOf, or oneOf schema used as items schema in array', () => {
+    const spec = yaml.safeLoad(
+      fs.readFileSync(
+        'test/cli-validator/mockFiles/oas3/composeModelItems.yaml'
+      )
+    );
+
+    const res = validate({ jsSpec: spec, isOAS3: true }, config);
+    expect(res.errors.length).toEqual(3);
+    expect(res.errors[0].message).toEqual('oneOf should be an array');
+    expect(res.errors[0].path).toEqual([
+      'components',
+      'schemas',
+      'one_of_array',
+      'items',
+      'oneOf'
+    ]);
+    expect(res.errors[1].message).toEqual('allOf should be an array');
+    expect(res.errors[1].path).toEqual([
+      'components',
+      'schemas',
+      'all_of_array',
+      'items',
+      'allOf'
+    ]);
+    expect(res.errors[2].message).toEqual('anyOf should be an array');
+    expect(res.errors[2].path).toEqual([
+      'components',
+      'schemas',
+      'any_of_array',
+      'items',
+      'anyOf'
+    ]);
+  });
+
```

Comment:
Will you add one positive test? It can just be one compose model which correct usage in one of the files